demo/site/src/app/[domain]/[language]/news/page.tsx

```diff
@@ -1,28 +1,38 @@
+"use server";
+
 import { gql, previewParams } from "@comet/cms-site";
-import { GQLNewsContentScopeInput } from "@src/graphql.generated";
+import { GQLNewsCategory, GQLNewsContentScopeInput } from "@src/graphql.generated";
 import { NewsList } from "@src/news/NewsList";
 import { newsListFragment } from "@src/news/NewsList.fragment";
 import { createGraphQLFetch } from "@src/util/graphQLClient";
 
 import { GQLNewsIndexPageQuery, GQLNewsIndexPageQueryVariables } from "./page.generated";
 
-export default async function NewsIndexPage({ params }: { params: { domain: string; language: string } }) {
-    // TODO support multiple domains, get domain by Host header
-    const { scope, previewData } = (await previewParams()) || { scope: { domain: params.domain, language: params.language }, previewData: undefined };
-    const graphqlFetch = createGraphQLFetch(previewData);
+export default async function NewsIndexPage({ params: { domain, language } }: { params: { domain: string; language: string } }) {
+    const scope = ((await previewParams())?.scope || { domain, language }) as GQLNewsContentScopeInput;
+    const newsList = await fetchNewsList(scope);
```

Comment:
It's nice that we can use the same function for the initial fetching on the server and the re-fetching on the client.